taskweaver/memory/experience.py

```diff
@@ -78,10 +74,14 @@ def __init__(
 
         self.experience_list: List[Experience] = []
 
-        self.exception_message_for_refresh = (
-            "Please cd to the `script` directory and "
-            "run `python -m experience_mgt --refresh` to refresh the experience."
-        )
+        self.experience_dir = None
+        self.sub_path = None
+
+    def set_experience_dir(self, experience_dir: str):
+        self.experience_dir = experience_dir
+
+    def set_sub_path(self, sub_path: str):
+        self.sub_path = sub_path
```

Comment:
is there any more specific meaning of the `dir` and `sub_path`? referring these storage related concept when querying experiences is not quite clear semantically.

Besides, what is the lifetime scope of the generator? It seems to be bound to a specific experience set within a `(dir, sub_path)` tuple, and reloads frequently whenever this tuple changes?

Reply:
1. I currently organize the experiences using sub directories. So, the experience_dir is the base dir and sub_path is the relative sub path under the base dir. 
2. Each generator is associated with a corresponding role. For example, the planner and code interpreter have two separate instances of the generator. Inside the generator, there is already `load_experience`. Here, I add this to make it easier for each role to load its own experiences based on its config.